Source/AssetRipper.Processing/AnimationClips/AnimationClipConverter.cs

```diff
@@ -524,41 +529,62 @@ private void AddPPtrKeyframe(in CurveData curveData, IAnimationClipBindingConsta
 				m_pptrs.Add(curveData, curve);
 			}
 
-			IPPtr_Object value = bindings.PptrCurveMapping[index];
+			IPPtr_Object value = m_clip.ClipBindingConstant_C74!.PptrCurveMapping[index];
 			IPPtrKeyframe key = curve.Curve.AddNew();
 			key.Time = time;
 			key.Value.CopyValues(value, new PPtrConverter(m_clip));
 		}
 
-		private static void GetPreviousFrame(IReadOnlyList<StreamedFrame> streamFrames, int curveID, int currentFrame, out int frameIndex, out int curveIndex)
+		private IGenericBinding GetBinding(int index)
 		{
-			for (frameIndex = currentFrame - 1; frameIndex >= 0; frameIndex--)
+			if (m_bindingsCache.TryGetValue(index, out IGenericBinding? binding))
 			{
-				StreamedFrame frame = streamFrames[frameIndex];
-				for (curveIndex = 0; curveIndex < frame.Curves.Length; curveIndex++)
+				return binding;
+			}
+			int curves = 0;
+			AccessListBase<IGenericBinding> bindings = m_clip.ClipBindingConstant_C74!.GenericBindings;
+			for (int i = 0; i < bindings.Count; i++)
+			{
+				binding = bindings[i];
+				if (binding.GetClassID() == ClassIDType.Transform)
 				{
-					StreamedCurveKey curve = frame.Curves[curveIndex];
-					if (curve.Index == curveID)
+					curves += binding.TransformType().GetDimension();
+				}
+				else
+				{
+					curves += 1;
+				}
+				if (curves > index)
+				{
+					m_bindingsCache[index] = binding;
+					if (binding.IsTransform() && binding.TransformType().GetDimension() < 1)
 					{
-						return;
+						// avoid infinite FOR loop when processing Transform animations
```

Comment:
Did you ever encounter this?

Reply:
no, but in case of processing a malformed animation/malformed assetbundle, I wanted to prevent the program from freezing in an infinite loop later (some iterators in `ProcessStreams()` rely on `binding.TransformType().GetDimension()` to advance)

Reply:
```cs
// If an animation was malformed, this avoids the possibility of an infinite FOR loop when processing Transform bindings.
```
How do you feel about this wording?

Reply:
sounds good, updating it in a second